### PR TITLE
Fix windows tests 3

### PR DESCRIPTION
**Pull request by erikbaranowski**
#### PR Checklist

- [ ] CHANGELOG.md updated
- [ ] Documentation added
- [x] Tests updated
- [x] Config converters updated

### CODE REVIEW

File: component/common/loki/wal/watcher_test.go [hunk 1]
```diff
@@ -685,9 +685,9 @@ func TestWatcher_StopAndDrainWAL(t *testing.T) {
 		watcher.Drain()
 		watcher.Stop()
 
-		// expecting 15s (missing 15 entries * 1 sec delay in AppendEntries) +/- 1.1s (taking into account the drain timeout
+		// expecting 15s (missing 15 entries * 1 sec delay in AppendEntries) +/- 2.0s (taking into account the drain timeout
 		// has one extra second.
-		require.InDelta(t, time.Second*15, time.Since(now), float64(time.Millisecond*1100), "expected the drain procedure to take around 15s")
+		require.InDelta(t, time.Second*15, time.Since(now), float64(time.Millisecond*2000), "expected the drain procedure to take around 15s")
```
Comment (erikbaranowski) [inline component/common/loki/wal/watcher_test.go:690]: @thepalbi These failures are sporadic for windows testing. If you feel this should be done a different way to make them more resilient let me know.

Comment (mattdurham) [reply component/common/loki/wal/watcher_test.go:690]: file io tests on windows are always a flaky. Sounds reasonable but if it gets to flaky we may have to make it linux only.

**Review by mattdurham — APPROVED** [review]
lgtm

File: component/common/loki/wal/watcher_test.go (hunk 1 above)
Comment (thepalbi) [reply component/common/loki/wal/watcher_test.go:690]: I'll take a look after to see if there's a way to make them more resilient

**Pull request merged**